repos/keg-components/src/components/form/checkbox/checkbox.wrapper.js

```diff
@@ -35,19 +35,27 @@ const useCheckedState = (isChecked, themeStyles) => {
 }
 
 /**
- * Sets / Updates the value of the checked Component
- * Calls the callback, if it's passed in
+ * Returns a memoized callback that sets / updates the value of the checked Component and 
+ * calls the callback, if it's passed in
  * @param {boolean} isChecked - Current state of the check value
  * @param {function} setChecked - Update the checked state
  * @param {function} onChange - Callback to call when the state changes
+ * @param {boolean} options.disableCheck - `setChecked` can only be called with `true` if this value is false
+ * @param {boolean} options.disableUncheck - `setChecked` can only be called with `false` if this value is also false
  *
  * @returns {function} - The checked state update function
  */
-const setCheckedValue = (isChecked, setChecked, onChange) => {
-  return event => {
-    setChecked(!isChecked)
+const getCheckboxPressHandler = (
```

Comment:
This should be called `useCheckboxPressHandler` cause it's a hook

Reply:
Thanks, updated